cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceTransformExecutorFactory.java

```diff
@@ -80,7 +77,7 @@ public MapReduceTransformExecutorFactory(MapReduceTaskContext taskContext,
     Configuration hConf = hadoopContext.getConfiguration();
     this.mapOutputKeyClassName = hConf.get(ETLMapReduce.MAP_KEY_CLASS);
     this.mapOutputValClassName = hConf.get(ETLMapReduce.MAP_VAL_CLASS);
-    this.isMapper = hadoopContext instanceof Mapper.Context;
+    this.isMapPhase = hadoopContext instanceof Mapper.Context;
```

Comment:
should not set the parent variable in the child. the parent constructor should take this as another argument, and you should pass it in super().